govcd/edgegateway_test.go

```diff
@@ -743,3 +743,51 @@ func (vcd *TestVCD) TestListEdgeGateway(check *C) {
 	check.Assert(foundName, Equals, true)
 	check.Assert(foundHref, Equals, true)
 }
+
+func (vcd *TestVCD) Test_UpdateEdgeGateway(check *C) {
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip("Skipping test because no edge gateway given")
+	}
+	edge, err := vcd.vdc.GetEdgeGatewayByName(vcd.config.VCD.EdgeGateway, false)
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
+	var saveEGW = types.EdgeGateway{
+		Name:        edge.EdgeGateway.Name,
+		ID:          edge.EdgeGateway.ID,
+		Status:      edge.EdgeGateway.Status,
+		HREF:        edge.EdgeGateway.HREF,
+		Description: edge.EdgeGateway.Description,
+		Configuration: &types.GatewayConfiguration{
+			AdvancedNetworkingEnabled: edge.EdgeGateway.Configuration.AdvancedNetworkingEnabled,
+		},
+	}
+
+	newName := "UpdatedEGWName"
+	newDescription := "Updated description"
+
+	edge.EdgeGateway.Name = newName
+	edge.EdgeGateway.Description = newDescription
+
+	err = edge.Update()
+	check.Assert(err, IsNil)
+
+	edge, err = vcd.vdc.GetEdgeGatewayById(saveEGW.ID, true)
```

Comment:
I think `update` should also update referenced Instance, not only send data.
So `edge.Update` -> would means that after `edge.EdgeGateway.Name` I would see new name. I don't need 
`edge, err = vcd.vdc.GetEdgeGatewayById(saveEGW.ID, true)`
Or return updated instance - usually we do that.

Reply:
Added explicit refresh at the end of Update, as agreed